rules/do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds/rule.md

```diff
@@ -0,0 +1,109 @@
+---
+type: rule
+tips: ""
+title: Do you Optimize Your TinaCMS Project for Clarity, Performance, and
+  Reliable Builds
+seoDescription: Learn effective strategies to structure and optimize your
+  TinaCMS project for clarity, performance, and successful builds.
+uri: do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds
+authors:
+  - title: ""
+created: 2024-08-28T16:15:00.000Z
+guid: 3c59f56f-34b0-47a7-902a-718eb340d2e5
+---
+Structuring and optimizing your TinaCMS project is essential to achieve clarity, enhance performance, and prevent build failures. Poorly optimized projects can lead to slow site performance, increased server load, and even failed builds due to excessive or inefficient data requests.
+
+Let’s explore how to structure your project effectively and apply best practices to boost performance both in runtime and during the build process.
+
+## 1. Structuring Your TinaCMS Architecture
+
+When working with large datasets or generating multiple subcomponents, following best practices is crucial to maintain performance and clarity.
+
+### Bad Practices:
+
```

Comment:
Was this generated via the SSW Rules Writer?
<img width="578" alt="image" src="https://github.com/user-attachments/assets/f9a76288-4d1e-4d47-bf52-2957285ca4d2">

The reason I ask is there doesn't seem to be a break i.e. `<!--endintro-->` to separate the intro from the main content

Reply:
I did use SSW Rules Writter, but after iterations and mofidications I probably deleted that "endintro" by mistake.
It's fixed now.